diff --git a/node_modules/react-router-dom/index.d.ts b/node_modules/react-router-dom/index.d.ts
index 146c97d..27ff56f 100644
--- a/node_modules/react-router-dom/index.d.ts
+++ b/node_modules/react-router-dom/index.d.ts
@@ -74,7 +74,8 @@ export declare const NavLink: React.ForwardRefExoticComponent<NavLinkProps & Rea
  * use in our exported `<Link>`.
  */
 export declare function useLinkClickHandler<E extends Element = HTMLAnchorElement>(to: To, { target, replace: replaceProp, state }?: {
-    target?: React.HTMLAttributeAnchorTarget;
+  // @ts-ignore
+  target?: React.HTMLAttributeAnchorTarget;
     replace?: boolean;
     state?: any;
 }): (event: React.MouseEvent<E, MouseEvent>) => void;
